server/src/main/java/org/elasticsearch/index/codec/vectors/es818/DirectIOLucene99FlatVectorsFormat.java

```diff
@@ -67,50 +63,14 @@ public FlatVectorsWriter fieldsWriter(SegmentWriteState state) throws IOExceptio
         return new Lucene99FlatVectorsWriter(state, vectorsScorer);
     }
 
-    private static final IOContext DIRECT_IO_CONTEXT = new IOContext() {
-        @Override
-        public Context context() {
-            return Context.DEFAULT;
-        }
-
-        @Override
-        public MergeInfo mergeInfo() {
-            return null;
-        }
-
-        @Override
-        public FlushInfo flushInfo() {
-            return null;
-        }
-
-        @Override
-        public Set<FileOpenHint> hints() {
-            return Set.of(DirectIOHint.INSTANCE);
-        }
-
-        @Override
-        public IOContext withHints(FileOpenHint... hints) {
-            return this;
-        }
-
-        @Override
-        public Optional<ReadAdvice> readAdvice() {
-            return Optional.empty();
-        }
-
-        @Override
-        public IOContext withReadAdvice(ReadAdvice advice) {
-            return this;
-        }
-    };
-
     @Override
     public FlatVectorsReader fieldsReader(SegmentReadState state) throws IOException {
+        var newHints = Stream.concat(state.context.hints().stream(), Stream.of(DirectIOHint.INSTANCE)).toArray(FileOpenHint[]::new);
         SegmentReadState directIOState = new SegmentReadState(
             state.directory,
             state.segmentInfo,
             state.fieldInfos,
-            DIRECT_IO_CONTEXT,
+            state.context.withHints(newHints),
```

Comment:
These hints will just be overridden within the constructor of `Lucene99FlatVectorsReader`. We need a new `IOContext` implementation passed in here that adds `DirectIOHint` to any hints that are specified by `withHints`, and returns the modified hints from the `IOContext.hints()` method

Reply:
D'oh! I see.